tests/performance/run_server_performance_test.py

```diff
@@ -0,0 +1,280 @@
+                        help="The URL of the product to store the results "
+                             "for, in the format of host:port/ProductName.")
+    parser.add_argument('-o', '--output',
+                        type=str,
+                        required=True,
+                        help="Output file name for printing statistics.")
+    parser.add_argument('-u', '--users',
+                        type=int,
+                        default=1,
+                        help="Number of users")
+    parser.add_argument('-v', '--verbose',
+                        action='store_true',
+                        help="Print the output of CodeChecker commands.")
+
+    return parser.parse_args()
+
+
+class StatManager:
+    """
+    This class stores the statistics of the single user events and prints them
+    in CSV format. To produce a nice output the users should do the same tasks
+    in the same order, e.g. they should all store, query and delete a run
+    in this order. In the output table a row belongs to each user. The columns
+    are the durations of the accomplished tasks.
+    """
+
+    def __init__(self):
+        # In this dictionary user ID is mapped to a list of key-value
+        # pairs: the key is a process name the value is its duration.
+        self._stats = defaultdict(list)
+
+    def add_duration(self, user_id, task_name, duration):
+        """
+        Add the duration of an event to the statistics.
+        """
+        self._stats[user_id].append((task_name, duration))
+
+    def print_stats(self, file_name):
+        if not self._stats:
+            return
+
+        with open(file_name, 'w') as f:
+            writer = csv.writer(f)
+
+            _, durations = self._stats.iteritems().next()
+            header = ['User'] + map(lambda x: x[0], durations)
+
+            writer.writerow(header)
+
+            for user_id, durations in self._stats.iteritems():
+                writer.writerow([user_id] + map(lambda x: x[1], durations))
+
+
+class UserSimulator:
+    """
+    This class simulates a user who performs actions one after the other. The
+    durations of the single actions are stored in the statistics.
+    """
+
+    _counter = 0
+
+    def __init__(self, stat):
+        UserSimulator._counter += 1
+        self._id = UserSimulator._counter
+        self._actions = list()
+        self._stat = stat
+
+    def get_id(self):
+        return self._id
+
+    def add_action(self, name, func, args):
+        """
+        This function adds a user action to be played later.
+        name -- The name of the action to identify it in the statistics output.
+        func -- A function object on which @return_duration decorator is
+                applied.
+        args -- A tuple of function arguments to be passed to func.
+        """
+        self._actions.append((name, func, args))
+
+    def play(self):
+        for name, func, args in self._actions:
+            self._user_random_sleep()
```

Comment:
This random sleep time should follow exponential distribution and its mean should be configurable from command line.
"0" should mean that we do not wait anything (0s no random wait) between generating events.